avm/src/vm/opcode.rs

```diff
@@ -3076,18 +3117,86 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
-
-      let bind = Server::try_bind(&addr);
-      match bind {
-        Ok(server) => {
-          let server = server.serve(make_svc);
+      match &Program::global().http_config {
+        HttpType::HTTP(http) => {
+          let port_num = http.port;
+          let addr = SocketAddr::from(([0, 0, 0, 0], port_num));
+          let make_svc = make_service_fn(|_conn| async { Ok::<_, Infallible>(service_fn(http_listener)) });
+
+          let bind = Server::try_bind(&addr);
+          match bind {
+            Ok(server) => {
+              let server = server.serve(make_svc);
+              tokio::spawn(async move { server.await });
+              println!("HTTP server listening on port {}", port_num);
+            },
+            Err(ee) => eprintln!("HTTP server failed to listen on port {}: {}", port_num, ee),
+          }
+        },
+        HttpType::HTTPS(https) => {
+          let port_num = https.port;
+          let addr = SocketAddr::from(([0, 0, 0, 0], port_num));
+          let tls_cfg = {
+            let certs = pemfile::certs(
+              &mut BufReader::new(base64::decode(https.cert_b64.as_str()).unwrap().as_slice())
+            );
+            let certs = match certs {
+              Ok(certs) => certs,
+              Err(_) => panic!("Failed to load certificate"),
+            };
+            eprintln!("{:?}", certs);
+            let key = {
+              let keys = pemfile::pkcs8_private_keys(
+                &mut BufReader::new(
+                  base64::decode(https.priv_key_b64.as_str()).unwrap().as_slice()
+                )
+              );
+              let keys = match keys {
+                Ok(keys) => keys,
+                Err(_) => panic!("Failed to load private key"),
+              };
+              if keys.len() != 1 {
+                eprintln!("{:?}", keys);
+                panic!("Expected a single private key");
+              }
+              keys[0].clone()
+            };
+            let mut cfg = rustls::ServerConfig::new(rustls::NoClientAuth::new());
+            cfg.set_single_cert(certs, key)
+              .map_err(|e| panic!(e)).unwrap();
+            cfg.set_protocols(&[b"h2".to_vec(), b"http/1.1".to_vec()]);
+            Arc::new(cfg)
+          };
+          let tcp = TcpListener::bind(&addr).await;
+          let tcp = match tcp {
+            Ok(tcp) => tcp,
+            Err(e) => panic!(e),
+          };
+          let tls_acceptor = TlsAcceptor::from(tls_cfg);
+          let incoming_tls_stream = stream! {
+            loop {
+              let accept = tcp.accept().await;
+              if accept.is_err() { continue; }
+              let accept = match accept {
+                Ok(accept) => accept,
+                Err(e) => {
+                  eprintln!("TLS Error: {:?}", e);
+                  continue;
+                },
+              };
+              let (socket, _) = accept;
```

Comment:
all of this can be reduced to:
```rust
let socket = match tcp.accept().await {
  Ok((socket, _remote_addr)) => socket,
  Err(e) => {
    eprintln!("TLS Error: {:?}", e); // I assume you want this, although currently you skip this on line 3179
    continue;
  }
};
```

Reply:
Line 3179 was added because I got the HTTPS server in a weird stuck state when I let it fall through to line 3183. I'd probably keep the `is_err()` check and just unwrap here.